modules/apps/kate.nix

```diff
@@ -0,0 +1,89 @@
+{ config, lib, ... }:
+
+let
+  # compute kate's magic TabHandlingMode
+  boolToInt = b: if b then 1 else 0;
+  t = boolToInt config.programs.kate.editor.indent.tabFromEverywhere;
+  s = boolToInt config.programs.kate.editor.indent.undoByShiftTab;
+  tabHandlingMode = t + 2 * s - 1;
+in
+{
+  options.programs.kate.editor = {
+    tabWidth = lib.mkOption {
+      description = "The width of a single tab (\\t) sign (in number of spaces).";
```

Comment:
Shouldn't it just be \t here and not \\\t?

Reply:
`\t` is rendered to the tab whitespace.
`\''t` is rendered to the tab character. I switched to that.

Source: [the nix documentation](https://nixos.org/manual/nix/stable/language/values.html#type-string)